python/lsst/sims/utils/timeTransformations.py

```diff
@@ -0,0 +1,195 @@
+"""
+For definitions of all (or, at least, most of) the time systems handled in this file, see
+
+https://www.cv.nrao.edu/~rfisher/Ephemerides/times.html
+"""
+
+import numpy as np
+import warnings
+import palpy
+from lsst.sims.utils import Ut1MinusUtcData
+from lsst.sims.utils import TT_from_TAI_Data
+
+__all__ = ["taiFromUtc", "utcFromTai",
+           "dutFromUtc",
+           "ut1FromUtc", "utcFromUt1",
+           "dttFromUtc", "ttFromTai",
+           "tdbFromTt"]
+
+
+def taiFromUtc(utc):
+    """
+    Use the palpy method palDat to convert from Coordinated Universal Time
+    (UTC) to International Atomic Time (TAI)
+
+    @param [in] utc is the UTC time as an MJD
+
+    @param [out] TAI time as an MJD
+    """
+
+    dt = palpy.dat(utc) # returns TAI-UTC in seconds
+    return utc + dt/86400.0
+
+
+def utcFromTai(tai):
+    """
+    Use the palpy method palDat to convert from International Atomic Time
+    (TAI) to Coordinated Universal Time (UTC)
+
+    @param [in] tai is the TAI time as an MJD
+
+    @param [out] utc time as an MJD
+    """
+
+    # because the PALPY method only returns TAI-UTC as a function
+    # of UTC, we will have to construct an array of TAI and UTC
+    # values using taiFromUtc and interpolate along them
+
+    sec_to_day = 1.0/86400.0
+
+    dt_approx = palpy.dat(tai)*sec_to_day
+
+    if dt_approx == 0.0:
+        return tai
+
+    utc_arr = np.arange(tai - 1.0*dt_approx, tai+1.0*dt_approx, 1.0e-6)
+    tai_arr = np.array([taiFromUtc(utc) for utc in utc_arr])
+
+    return np.interp(tai, tai_arr, utc_arr)
+
+
+def dutFromUtc(utc):
+    """
+    Use data downloaded from
+
+    ftp://cddis.gsfc.nasa.gov/pub/products/iers/
+
+    to return UT1-UTC as a function of UTC
+
+    @param [in] UTC as an MJD
+
+    @param [out] UT1-UTC in seconds
+    """
+
+    if utc<Ut1MinusUtcData._mjd_arr[0] or utc>Ut1MinusUtcData._mjd_arr[-1]:
+        msg = "UTC = %e is outside of the the bounds " % utc \
+               + "for which we have UT1-UTC "\
+               + "data (%e <= utc <= %e)\n" % (Ut1MinusUtcData._mjd_arr[0], Ut1MinusUtcData._mjd_arr[-1]) \
+               + "We will return UT1-UTC = 0, for lack of a better idea"
+        warnings.warn(msg)
+
+        return 0.0
+
+    return np.interp(utc, Ut1MinusUtcData._mjd_arr, Ut1MinusUtcData._dut_arr)
```

Comment:
Wait - is this an interpolation of the UT1-UTC values? I thought they were discontinuous (i.e. leap seconds)?